Sources/Accord.Math/Optimization/QuadraticObjectiveFunction.cs

```diff
@@ -174,18 +191,21 @@ public double ConstantTerm
         /// 
         public QuadraticObjectiveFunction(double[,] quadraticTerms, double[] linearTerms, params string[] variables)
         {
-            if (quadraticTerms.GetLength(0) != quadraticTerms.GetLength(1))
+            if (!quadraticTerms.IsSquare())
                 throw new DimensionMismatchException("quadraticTerms", "The matrix must be square.");
 
-            if (quadraticTerms.GetLength(0) != linearTerms.Length)
+            if (!quadraticTerms.IsSymmetric(rtol: 1e-6))

```

Comment:
I think this is fine. If for some reason it throws for a matrix that was indeed supposed to be symmetric, the user can always "force" it to be symmetric with [GetSymmetric](http://accord-framework.net/docs/html/M_Accord_Math_Matrix_GetSymmetric__1_1.htm) or manually.

Reply:
Cool - my thinking was similar and I had spotted that method - would work well. Another pair of methods that would be useful would be `GetSymmetricPart()` and `GetAntiSymmetricPart` evaluating `0.5(M+M')` and `0.5(M-M')` respectively.